net/mwan3/files/lib/mwan3/mwan3.sh

```diff
@@ -352,6 +352,46 @@ mwan3_set_custom_ipset()
 	error=$(echo "$update" | $IPS restore 2>&1) || LOG error "set_custom_ipset: $error"
 }
 
+mwan3_load_ipset_rule()
+{
+	local family
+	local rule=$1
+
+	config_get family $rule family 'any'
+	local dir ips
+	local has_ipv6=0
+	local has_ipv4=0
+	for dir in src dest; do
+		config_get ips $rule ${dir}_ip
+		if [ -z "$ips" ] || [ -n "${ips##*,*}" ]; then
+			continue
+		fi
+		if [ "$family" = any ]; then
+			mwan3_push_update create mwan3_rule_${rule}_${dir} list:set
+			mwan3_push_update create mwan3_rule_${rule}_${dir}_ipv4 hash:ip
+			mwan3_push_update create mwan3_rule_${rule}_${dir}_ipv6 hash:ip family inet6
```

Comment:
if use ipset, src ip may be `ip/cidr`, so we may use hash:net
and please set the ipset hash size, default is 1024, may eat memory size.

Reply:
How much memory do you think a 1024 hash uses?  I changed this (and elsewhere in the codebase that uses hash:net), but the overall memory footprint will be unnoticeable. I saw elsewhere you are using luci - if you are on a very restricted memory device, you may consider turning off uhttpd, which will save a fair bit of RAM.